server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/ConsistencyTasksIntegrationTest.java

```diff
@@ -96,6 +104,14 @@ private TestingSessionProbe(TestingSession testingSession) {
         public TestingSession getTestingSession() {
             return testingSession;
         }
+
+        public List<ComposedMessageId> listMessagesInTruthTable() {
```

Comment:
why do we need to cheat that way?

Reply:
Because we need a scenario where the insertion fails in one table. I don't know if we have any way of retrieving the ComposeMessageId of the failed message (in order to use it in the assertion after), and so it leads to the listAll() here.

Reply:
I reviewed the two usages of this method and the problem relies on the way the tests are written.

When doing testing, we do mostly blackbox testing: we don't care how things are working, we just care about the result we can view from the outside.

Inconsistencies are things we observes as a users and users don't have access to "source of truth", they just do IMAP and JMAP.

So you should be able to write those tests without this method by using the right protocol and thinking about the problem you fix for the end user.

Reply:
(yes for sure however previously written test in that class did not follow that approach.)

To be noted: that was not accounted in the Definition Of Done, and likely would have had influenced the vote of team members about the task complexity.

So either we upgrade the **definition of done**, include end user experience for other inconsistency tasks in it, re-evaluate the task, and adapt sprint content (most likely by dropping tasks) or we live with that probe.

IMO I'm for option 2: this work is not related to sprint objectives and it would likely be what I would end up dropping.

Reply:
> (yes for sure however previously written test in that class did not follow that approach.)

I can only catch a small part of problems during reviews. Here I catch it because of a hack that was visible enough. I don't see any mean to prevent that as long as we rely so much on code reviews (vs more collaboration on writing code)

> To be noted: that was not accounted in the Definition Of Done, and likely would have had influenced the vote of team members about the task complexity.
> 
> So either we upgrade the **definition of done**, include end user experience for other inconsistency tasks in it, re-evaluate the task, and adapt sprint content (most likely by dropping tasks) or we live with that probe.

You seem to be against the first option and I'm obviously against the second one.

What about delaying the merge of this PR waiting for more time to make it good enough?

Reply:
I'm not against the first option, I'm just wondering where to find the time budget for it.

I don't want to put more pressure on my team with unplanned redefinition, as I don't think it is the solution.

I think we should systematically re-evaluate things when the definition of done evolves, this should be a good habit.

> What about delaying the merge of this PR waiting for more time to make it good enough?

I have a conceptual problem about doing things halthway. IMO either we do them all, or we do nothing.

Reply:
Isn't option 2 (living with the probe) doing things halfway? I depends if you talk for master source code or task management I guess